resources/shaders/dxr_shadow_functions.hlsl

```diff
@@ -1,14 +1,10 @@
-#include "rt_global.hlsl"
+#ifndef __RT_SHADOWS_FUNCTIONS__
+#define __RT_SHADOWS_FUNCTIONS__
 
-struct ShadowHitInfo
-{
-  float is_hit;
-  float thisvariablesomehowmakeshybridrenderingwork_killme;
-};
-
-struct Attributes { };
+#include "dxr_global.hlsl"
+#include "dxr_structs.hlsl"
 
-bool TraceShadowRay(uint idx, float3 origin, float3 direction, float far, unsigned int depth)
+bool TraceShadowRay(float3 origin, float3 direction, float far, uint ray_contr_idx, uint miss_idx, unsigned int depth)
```

Comment:
I prefer these variables to be at the end and defaulted but :shrugh:

Reply:
I'd rather not default these variables. Users need to think about their miss shader and hit groups structures and put the right index in there.